staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -511,6 +511,13 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 		return newErrWatcher(err), nil
 	}
 
+	// FIXME: Change the condition to ensure that we're not allocating memory for initEvents
+	//   without any reasonable boundary.
+	if false {
+		// TODO: See above - can 429 be returned as error - or the above argument holds here too?
+		return nil, errors.NewTooManyRequests("<TODO>")
```

Comment:
@liggitt - can you please share your opinion about this approach?
[This is the skeleton - if we agree that it's a reasonable approach, I will work on that, add steps, etc.]